specification/ai/Azure.AI.Projects/servicepatterns.tsp

```diff
@@ -0,0 +1,123 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@azure-tools/typespec-azure-core";
+
+namespace Azure.AI.Projects.ServicePatterns {
+  @Rest.autoRoute
+  interface VersionedOperations<
+    TEntityType extends Reflection.Model,
+    TListLatestParameters extends Reflection.Model = {},
+    TListVersionsParameters extends Reflection.Model = {}
+  > {
+    #suppress "@azure-tools/typespec-azure-core/verb-conflict" "We are using the resource action as a building block."
+    @doc("List all versions of the given {name}", TEntityType)
+    @Rest.action("versions")
+    @Rest.actionSeparator("/")
+    @Http.get
+    listVersions is BuildingBlocks.CoreOps.ResourceAction<
+      TEntityType,
+      TListVersionsParameters,
+      Azure.Core.Page<TEntityType>
+    >;
+
+    @doc("List the latest version of each {name}", TEntityType)
+    @Rest.listsResource(TEntityType)
+    listLatest is Azure.Core.Foundations.ResourceList<
+      TEntityType,
+      TListLatestParameters,
+      Azure.Core.Page<TEntityType>
+    >;
```

Comment:
"list" operations should look similar across routes /dataset, /deployment, /connections, /indexes - Option to order_by (?) if we keep systemData. For /dataset I see we have input argument order_by and others ( `top`, skip, tags) in the list_versions method, but I don't see them on the list_latest method. Since these methos both list versions, I expect them to accept similar inputs. Any reason why we don't have order_by on list_latest?